File: 30 Days of Code/Day 25 Running Time and Complexity/Solution.cs
using System;
using System.Linq;

class Solution {

    static bool isEven( int number ) {
        return number % 2 == 0;
    }

    static void Main( String[ ] args ) {
        int T = Int32.Parse( Console.ReadLine( ) );

        for( int idx = 1; idx <= T; idx++ ) {
            int n = Int32.Parse( Console.ReadLine( ) );
            bool isPrime = true;

            if( n == 2 || n == 3 || n == 5 || n == 7 ) {
                isPrime = true;
            } else if( isEven( n ) || n == 1 || n == 9 ) {
                isPrime = false;
            } else {
                int sqrt = Convert.ToInt32( Math.Sqrt( n ) );

                for( int jdx = 3; ( jdx <= sqrt && isPrime ); jdx += 2 ) {
                    if( ( n % jdx ) == 0 ) {
                        isPrime = false;
                    }
                }
            }

            if( isPrime ) {
                Console.WriteLine( "Prime" );
            } else {
                Console.WriteLine( "Not prime" );
            }
            
        }
    }
}

File: 30 Days of Code/Day 25 Running Time and Complexity/Solution.py
import math

def isEven( number ):
    return ( number % 2 ) == 0

def isPrime( number ):
    if number == 2:
        return True
    if number == 1 or number == 9 or isEven( number ):
        return False
    idx = 1
    for x in range( 3, math.ceil( math.sqrt( number ) ) + 1, 2):
        if number % x == 0:
            return False
        idx += 1
        
    print( idx )

    return True

T = int( input( ).strip( ) )
for idx in range( T ):
    n = int( input( ).strip( ) )
    message = "Prime" if isPrime( n ) else "Not prime"
    print( message )
